en/Data Structures/Queues/queue.md

```diff
@@ -0,0 +1,49 @@
+# Queue
+
+## Description:
+A queue is a linear data structure that follows the First-In-First-Out (FIFO) principle. It is often compared to a real-world queue of people waiting in line. The element that is added first is the one that gets removed first. Queues are commonly used for various applications, such as task scheduling, managing requests, and more.
+
+### Time Complexity:
```

Comment:
These complexities only apply to particular implementations (e.g. linked list, array with space reserved at both ends). This should be noted somewhere. This section should be preceded by a section detailing the abstract "queue" interface: What operations are there? (Initialization of an empty queue, enqueue, dequeue, peek), and what do they do? (It's mostly in the name, but a sentence or two wouldn't hurt.)